yarn-project/aztec.js/src/aztec_rpc_client/wallet.ts

```diff
@@ -108,6 +111,56 @@ export class EntrypointWallet extends BaseWallet {
   }
 }
 
+/**
+ * A wallet implementation supporting EIP1271.
+ * This wallet inserts eip1271-like witnesses into the RPC, which are then fetched using an oracle
+ * to provide authentication data to the contract during execution.
+ */
+export class EipEntrypointWallet extends BaseWallet {
+  constructor(rpc: AztecRPC, protected accountImpl: Eip1271AccountEntrypoint) {
+    super(rpc);
+  }
+
+  /**
+   * Create a transaction request and add the eip1271 witness to the RPC.
+   * Note:  When used in simulations, the witness that is inserted could be used later by attacker with
+   *        access to the RPC.
+   *        Meaning that if you were to use someone elses rpc with db you could send these transactions.
+   *        For simulations it would be desirable to bypass such that no data is generated.
+   *
+   * @param executions - The function calls to execute.
+   * @param opts - The options.
+   * @returns - The TxRequest
+   */
+  async createTxExecutionRequest(
+    executions: FunctionCall[],
+    opts: CreateTxRequestOpts = {},
+  ): Promise<TxExecutionRequest> {
+    const { txRequest, message, witness } = await this.accountImpl.createTxExecutionRequestWithWitness(
+      executions,
+      opts,
+    );
+    await this.rpc.addEip1271Witness(Fr.fromBuffer(message), witness);
+    return txRequest;
+  }
+
+  sign(messageHash: Buffer): Promise<Buffer> {
+    return Promise.resolve(this.accountImpl.sign(messageHash));
+  }
+
+  /**
+   * Signs the `messageHash` and adds the witness to the RPC.
+   * This is useful for signing messages that are not directly part of the transaction payload, such as
+   * approvals .
+   * @param messageHash - The message hash to sign
+   */
+  async signAndAddEip1271Witness(messageHash: Buffer): Promise<void> {
+    const witness = await this.accountImpl.createEip1271Witness(messageHash);
+    await this.rpc.addEip1271Witness(Fr.fromBuffer(messageHash), witness);
+    return Promise.resolve();
+  }
```

Comment:
Should we return the signed message as well? I'm thinking this could be used for "exporting" the 12721 auth message and sharing it with another user who embeds it in their tx.

Reply:
You could use the sign for that, get the sig, then on the other side he recreate it 🤷 but I follow what you are saying, might be fine to add to retrieve directly.